src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -26,15 +63,26 @@ public boolean isFull() {
     public int top() {
         if (isEmpty())
             return -1;
-        return last.value;
+        return getLast().getValue();
     }
 
     public int pop() {
         if (isEmpty())
-            return -1;
-        int ret = last.value;
-        last = last.prev;
+            return EMPTY_REPRESENTATION;
+        int ret = getLast().getValue();
+        setLast(getLast().getPrev());
         return ret;
     }
 
+    Node getLast() {
```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.